ESP32S3 build mcuboot error

**Describe the bug**
 An error occurred when I compiled mcuboot.  error is "fatal error: cmsis_core.h: No such file or directory #include <cmsis_core.h>".

**To Reproduce**
Steps to reproduce the behavior:
1. cd zephyr
2. west build -b esp32s3_devkitm --sysbuild samples/hello_world

**Expected behavior**
build the mcuboot  successful.

**Logs and console output**
.....
Loading Zephyr default modules (Zephyr base).
-- Application: D:/zephyr3/bootloader/mcuboot/boot/zephyr
-- CMake version: 3.22.1
-- Found Python3: C:/Python38/python.exe (found suitable version "3.8.10", minimum required is "3.8") found components: Interpreter
-- Cache files will be written to: D:/zephyr3/zephyr/.cache
-- Zephyr version: 3.4.99 (D:/zephyr3/zephyr)
-- Found west (found suitable version "1.0.0", minimum required is "0.14.0")
-- Board: esp32s3_devkitm
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.1 (C:/Users/sam/zephyr-sdk-0.16.1)
-- Found toolchain: zephyr 0.16.1 (C:/Users/sam/zephyr-sdk-0.16.1)
-- Found Dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: D:/zephyr3/zephyr/boards/xtensa/esp32s3_devkitm/esp32s3_devkitm.dts
-- Found devicetree overlay: D:/zephyr3/bootloader/mcuboot/boot/zephyr/app.overlay
-- Generated zephyr.dts: D:/zephyr3/zephyr/build/mcuboot/zephyr/zephyr.dts
-- Generated devicetree_generated.h: D:/zephyr3/zephyr/build/mcuboot/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: D:/zephyr3/zephyr/build/mcuboot/zephyr/dts.cmake
Parsing D:/zephyr3/bootloader/mcuboot/boot/zephyr/Kconfig
Loaded configuration 'D:/zephyr3/zephyr/boards/xtensa/esp32s3_devkitm/esp32s3_devkitm_defconfig'
Merged configuration 'D:/zephyr3/bootloader/mcuboot/boot/zephyr/prj.conf'
Merged configuration 'D:/zephyr3/bootloader/mcuboot/boot/zephyr/boards/esp32s3_devkitm.conf'
Merged configuration 'D:/zephyr3/zephyr/build/mcuboot/zephyr/.config.sysbuild'
....
[11/16] Performing build step for 'mcuboot'
[4/175] Generating include/generated/version.h
-- Zephyr version: 3.4.99 (D:/zephyr3/zephyr), build: zephyr-v3.4.0-3278-g9a9855f711d0
[104/175] Building C object CMakeFiles/app.dir/main.c.obj
FAILED: CMakeFiles/app.dir/main.c.obj
.....
D:/zephyr3/bootloader/mcuboot/boot/zephyr/main.c:30:10: fatal error: cmsis_core.h: No such file or directory
   30 | #include <cmsis_core.h>
      |          ^~~~~~~~~~~~~~

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain :Zephyr SDK

--- a/include/zephyr/mgmt/mcumgr/mgmt/mgmt.h
+++ b/include/zephyr/mgmt/mcumgr/mgmt/mgmt.h
@@ -11,6 +11,7 @@
 #include <inttypes.h>
 #include <zephyr/sys/slist.h>
 #include <zephyr/mgmt/mcumgr/smp/smp.h>
+#include <zephyr/mgmt/mcumgr/mgmt/mgmt_defines.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -22,124 +23,6 @@ extern "C" {
  * @ingroup mcumgr
  * @{
  */
-
-/**
- * Used at end of MCUmgr handlers to return an error if the message size limit was reached,
- * or OK if it was not
- */
-#define MGMT_RETURN_CHECK(ok) ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE
-
-/** Opcodes; encoded in first byte of header. */
-enum mcumgr_op_t {
-	/** Read op-code */
-	MGMT_OP_READ		= 0,
-
-	/** Read response op-code */
-	MGMT_OP_READ_RSP,
-
-	/** Write op-code */
-	MGMT_OP_WRITE,
-
-	/** Write response op-code */
-	MGMT_OP_WRITE_RSP,
-};
-
-/**
- * MCUmgr groups. The first 64 groups are reserved for system level mcumgr
- * commands. Per-user commands are then defined after group 64.
- */
-enum mcumgr_group_t {
-	/** OS (operating system) group */
-	MGMT_GROUP_ID_OS	= 0,
-
-	/** Image management group, used for uploading firmware images */
-	MGMT_GROUP_ID_IMAGE,
-
-	/** Statistic management group, used for retieving statistics */
-	MGMT_GROUP_ID_STAT,
-
-	/** Settings management (config) group, used for reading/writing settings */
-	MGMT_GROUP_ID_SETTINGS,
-
-	/** Log management group (unused) */
-	MGMT_GROUP_ID_LOG,
-
-	/** Crash group (unused) */
-	MGMT_GROUP_ID_CRASH,
-
-	/** Split image management group (unused) */
-	MGMT_GROUP_ID_SPLIT,
-
-	/** Run group (unused) */
-	MGMT_GROUP_ID_RUN,
-
-	/** FS (file system) group, used for performing file IO operations */
-	MGMT_GROUP_ID_FS,
-
-	/** Shell management group, used for executing shell commands */
-	MGMT_GROUP_ID_SHELL,
-
-	/** User groups defined from 64 onwards */
-	MGMT_GROUP_ID_PERUSER	= 64,
-
-	/** Zephyr-specific groups decrease from PERUSER to avoid collision with upstream and
-	 *  user-defined groups.
-	 *  Zephyr-specific: Basic group
-	 */
-	ZEPHYR_MGMT_GRP_BASIC	= (MGMT_GROUP_ID_PERUSER - 1),
-};
-
-/**
- * MCUmgr error codes.
- */
-enum mcumgr_err_t {
-	/** No error (success). */
-	MGMT_ERR_EOK		= 0,
-
-	/** Unknown error. */
-	MGMT_ERR_EUNKNOWN,
-
-	/** Insufficient memory (likely not enough space for CBOR object). */
-	MGMT_ERR_ENOMEM,
-
-	/** Error in input value. */
-	MGMT_ERR_EINVAL,
-
-	/** Operation timed out. */
-	MGMT_ERR_ETIMEOUT,
-
-	/** No such file/entry. */
-	MGMT_ERR_ENOENT,
-
-	/** Current state disallows command. */
-	MGMT_ERR_EBADSTATE,
-
-	/** Response too large. */
-	MGMT_ERR_EMSGSIZE,
-
-	/** Command not supported. */
-	MGMT_ERR_ENOTSUP,
-
-	/** Corrupt */
-	MGMT_ERR_ECORRUPT,
-
-	/** Command blocked by processing of other command */
-	MGMT_ERR_EBUSY,
-
-	/** Access to specific function, command or resource denied */
-	MGMT_ERR_EACCESSDENIED,
-
-	/** Requested SMP MCUmgr protocol version is not supported (too old) */
-	MGMT_ERR_UNSUPPORTED_TOO_OLD,
-
-	/** Requested SMP MCUmgr protocol version is not supported (too new) */
-	MGMT_ERR_UNSUPPORTED_TOO_NEW,
-
-	/** User errors defined from 256 onwards */
-	MGMT_ERR_EPERUSER	= 256
-};
-
-#define MGMT_HDR_SIZE		8
 
 /** @typedef mgmt_alloc_rsp_fn
  * @brief Allocates a buffer suitable for holding a response.

--- a/include/zephyr/mgmt/mcumgr/mgmt/mgmt_defines.h
+++ b/include/zephyr/mgmt/mcumgr/mgmt/mgmt_defines.h
@@ -1,0 +1,150 @@
+/*
+ * Copyright (c) 2018-2021 mcumgr authors
+ * Copyright (c) 2022-2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MGMT_MGMT_DEFINES_
+#define H_MGMT_MGMT_DEFINES_
+
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief MCUmgr mgmt API
+ * @defgroup mcumgr_mgmt_api MCUmgr mgmt API
+ * @ingroup mcumgr
+ * @{
+ */
+
+/**
+ * Used at end of MCUmgr handlers to return an error if the message size limit was reached,
+ * or OK if it was not
+ */
+#define MGMT_RETURN_CHECK(ok) ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE
+
+/** Opcodes; encoded in first byte of header. */
+enum mcumgr_op_t {
+	/** Read op-code */
+	MGMT_OP_READ		= 0,
+
+	/** Read response op-code */
+	MGMT_OP_READ_RSP,
+
+	/** Write op-code */
+	MGMT_OP_WRITE,
+
+	/** Write response op-code */
+	MGMT_OP_WRITE_RSP,
+};
+
+/**
+ * MCUmgr groups. The first 64 groups are reserved for system level mcumgr
+ * commands. Per-user commands are then defined after group 64.
+ */
+enum mcumgr_group_t {
+	/** OS (operating system) group */
+	MGMT_GROUP_ID_OS	= 0,
+
+	/** Image management group, used for uploading firmware images */
+	MGMT_GROUP_ID_IMAGE,
+
+	/** Statistic management group, used for retieving statistics */
+	MGMT_GROUP_ID_STAT,
+
+	/** Settings management (config) group, used for reading/writing settings */
+	MGMT_GROUP_ID_SETTINGS,
+
+	/** Log management group (unused) */
+	MGMT_GROUP_ID_LOG,
+
+	/** Crash group (unused) */
+	MGMT_GROUP_ID_CRASH,
+
+	/** Split image management group (unused) */
+	MGMT_GROUP_ID_SPLIT,
+
+	/** Run group (unused) */
+	MGMT_GROUP_ID_RUN,
+
+	/** FS (file system) group, used for performing file IO operations */
+	MGMT_GROUP_ID_FS,
+
+	/** Shell management group, used for executing shell commands */
+	MGMT_GROUP_ID_SHELL,
+
+	/** User groups defined from 64 onwards */
+	MGMT_GROUP_ID_PERUSER	= 64,
+
+	/** Zephyr-specific groups decrease from PERUSER to avoid collision with upstream and
+	 *  user-defined groups.
+	 *  Zephyr-specific: Basic group
+	 */
+	ZEPHYR_MGMT_GRP_BASIC	= (MGMT_GROUP_ID_PERUSER - 1),
+};
+
+/**
+ * MCUmgr error codes.
+ */
+enum mcumgr_err_t {
+	/** No error (success). */
+	MGMT_ERR_EOK		= 0,
+
+	/** Unknown error. */
+	MGMT_ERR_EUNKNOWN,
+
+	/** Insufficient memory (likely not enough space for CBOR object). */
+	MGMT_ERR_ENOMEM,
+
+	/** Error in input value. */
+	MGMT_ERR_EINVAL,
+
+	/** Operation timed out. */
+	MGMT_ERR_ETIMEOUT,
+
+	/** No such file/entry. */
+	MGMT_ERR_ENOENT,
+
+	/** Current state disallows command. */
+	MGMT_ERR_EBADSTATE,
+
+	/** Response too large. */
+	MGMT_ERR_EMSGSIZE,
+
+	/** Command not supported. */
+	MGMT_ERR_ENOTSUP,
+
+	/** Corrupt */
+	MGMT_ERR_ECORRUPT,
+
+	/** Command blocked by processing of other command */
+	MGMT_ERR_EBUSY,
+
+	/** Access to specific function, command or resource denied */
+	MGMT_ERR_EACCESSDENIED,
+
+	/** Requested SMP MCUmgr protocol version is not supported (too old) */
+	MGMT_ERR_UNSUPPORTED_TOO_OLD,
+
+	/** Requested SMP MCUmgr protocol version is not supported (too new) */
+	MGMT_ERR_UNSUPPORTED_TOO_NEW,
+
+	/** User errors defined from 256 onwards */
+	MGMT_ERR_EPERUSER	= 256
+};
+
+#define MGMT_HDR_SIZE		8
+
+/**
+ * @}
+ */
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* MGMT_MGMT_DEFINES_H_ */

--- a/west.yml
+++ b/west.yml
@@ -281,7 +281,7 @@ manifest:
       groups:
         - crypto
     - name: mcuboot
-      revision: 11ecbf639d826c084973beed709a63d51d9b684e
+      revision: 9bf7ce8c5fe8152836a6e00bd4444153bd950342
       path: bootloader/mcuboot
     - name: mipi-sys-t
       path: modules/debug/mipi-sys-t

